### DISCUSSION TITLE
Wifi streamer example with app push demo

### DISCUSSION THREAD

**Discussion by Kiesmith** [bitcraze/discussions#648] [Q&A]
I am trying to make my drone fly autonomously with the ability to stream video to my computer. 

I did this by the two demos provided by bitcraze.
 I have programmed my AI deck with the wifi-streamer from the ai-deck repository. 
I have programmed my actual drone also with the app push demo (the one where you put your hand over the ranger deck and it makes the drone fly and hover). 

**Result**
Drone does connect to wifi and I can run my code which returns the streamed video to my computer which gets processed by an AI on my PC. However, as soon as put my hand over the top sensor, it flies but the wifi stops and it stops streaming. It is a bit dodgy because the hover mode might crash sometimes when I do it like this to begin with. 

**How do I combine the two so I get can both stream video out from the drone for my PC to pick up and handle and have the drone fly autonomously?**

**Comment by Kiesmith** [comment]
Since the drone technically already streams  information to the crazyflie client in the computer, is it possible to actually send the video frames there without needing to do anything special and somehow pull that video into the code? That way I could just focus on making it fly as it transmits video frames (if i have the radio dongle connected). To do that I am unsure what I would have to do.

**Comment by hmllr** [comment]
Does this happen every time? The WiFi module and the gap8 are a bit sensitive to voltage drops, so if your drone is very close to max weight and/or the battery is not the best anymore I would expect similar problems as you see on takeoff. I assume you have a flowdeck, multiranger and ai-deck mounted, which is quite some weight. I would suggest to try a slower takeoff and maybe the newest batteries you have (If you have some, maybe also try some not standard batteries - I personally usually use some tattu 350mAh batteries for such heavy loads, which age very fast but are great in the beginning). 
Sending the video stream over radio will only allow a very low framerate, so i wouldn't recommend that.
Hope this helps,
Hanna

**Reply by Kiesmith** [reply]
Hey Hannah, Thankyou for the advice. It has been a while so I will refresh my insight.

Before I go further on this, may I ask how you got yours all to work together? There indeed is a voltage drop and I suppose the drone is close to max weight since I have the **flow deck, multiranger deck, AI deck installed** on the **Crazyflie 2.1**. 

**Following these results from the drone:** 

- drone suddenly crashes for no reason
- drone may crash on take-off 

Even though beforehand, successful take-off has occurred. I have reason to believe this is due to the battery now being of poor quality. Sometimes the battery does not even dimmer, it just turns off when crashing. I am not sure on this part of the drone if that is to do with the firmware or the battery, charging the battery fixes the issue. 


**To give you further detail on how I programmed the craziefly:**

- I programmed the AI deck seperately (I just flashed the wifi streaming example to the AI deck).
- I flashed my modified version of the app push demo, the drone flies around and avoids collision and navigates freely instead of hovering and avoiding any object in proximity.

they weren't flashed together, I have 2 VMs to flash both of them separately. Did you do it this way also? I have not looked into the pin configurations of the boards so I sometimes wonder: **Do the deck pin configurations interrupt each other (hence making the drone stop flying or crash when I activate the wifi streaming example)? or is it really just a voltage drop?** Also I cant seem to connect to drone Wi-Fi when it is flying.

I would love your help on this.
Kind regards,
Kieran.

**Reply by Kiesmith** [reply]
let me correct my insight on the wifi streaming example issue:

- Wifi streaming example works, only when drone is stationary. 
- Wifi streaming example disconnects to computer when I activate my drone by moving my hand to the top flow sensor for drone activation (same way the drone initiates take-off in **app push demo**)
- Never really been able to connect to wifi streaming example when flying, unsure if the connection is not there or because the battery does not have long enough duration. 
- usually when take-off with wifi streaming example is on and active, drone likes to crash instead on take-off. 
- usually after the examples happen above (drone crash when laptop connect to drone wifi example on take-off), trying to connect to the drone Wi-Fi again takes several minutes.

**Comment by knmcguire** [comment]
Hi! So you are trying quite some advanced stuff that potentially makes it quite difficult to debug this all. It should be able to work theoretically but  how about taking some steps back first and slowly increasing the complexity of what you are trying to do:

Let's start with a simpler example first and add complexity to it one by one:

1.  Only attach flowdeck. Use the [crazyflie python library](https://www.bitcraze.io/documentation/repository/crazyflie-clients-python/master/) and the [motion commander example](https://github.com/bitcraze/crazyflie-lib-python/blob/master/examples/autonomy/motion_commander_demo.py) (remove perhaps all the other fancy maneuvers). How does the drone fly? Is it stable? Is there enough texture for hover.
2.  Only attach the flowdeck and multiranger. Try out the [python pushdemo](https://github.com/bitcraze/crazyflie-lib-python/blob/master/examples/multiranger/multiranger_push.py) . Same thing, how is the hover, how is it responding on your hand?
3.  Now remove the multiranger and add the aideck (so now flowdeck+aideck). Try the (simplified) motion commander demo again while streaming the data. Are you still able to connect to the wifi streaming? or is it disconnecting like before?
4.  Now add the multiranger back on (flow+multi+aideck). Try out the push demo now with the aideck streaming. How is the quality now?

I'm pretty sure that at one of these you'll probably get stuck, but at least we will know what to focus on.
